internal/operands/vm-console-proxy/reconcile_test.go

```diff
@@ -54,30 +67,234 @@ var _ = Describe("VM Console Proxy Operand", func() {
 	})
 
 	It("should return name correctly", func() {
-		name := v.Name()
+		name := operand.Name()
 		Expect(name).To(Equal(operandName), "should return correct name")
 	})
 
 	It("should return functions from reconcile correclty", func() {
-		functions, err := v.Reconcile(mockedRequest)
+		functions, err := operand.Reconcile(&request)
 		Expect(err).ToNot(HaveOccurred(), "should not throw err")
 		Expect(len(functions)).To(Equal(7), "should return correct number of reconcile functions")
 	})
+
+	It("should create vm-console-proxy resources", func() {
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		ExpectResourceExists(bundle.serviceAccount, request)
+		ExpectResourceExists(bundle.clusterRole, request)
+		ExpectResourceExists(bundle.clusterRoleBinding, request)
+		ExpectResourceExists(bundle.configMap, request)
+		ExpectResourceExists(bundle.service, request)
+		ExpectResourceExists(bundle.deployment, request)
+		ExpectResourceExists(newRoute(namespace, "vm-console-proxy"), request)
+	})
+
+	It("should remove cluster resources on cleanup", func() {
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		ExpectResourceExists(bundle.serviceAccount, request)
+		ExpectResourceExists(bundle.clusterRole, request)
+		ExpectResourceExists(bundle.clusterRoleBinding, request)
+		ExpectResourceExists(bundle.configMap, request)
+		ExpectResourceExists(bundle.service, request)
+		ExpectResourceExists(bundle.deployment, request)
+		ExpectResourceExists(newRoute(namespace, "vm-console-proxy"), request)
+
+		_, err = operand.Cleanup(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// ExpectResourceNotExists(bundle.serviceAccount, request)
```

Comment:
Can you please investigate what makes these calls fail? Commented code IMO is unclean.

Reply:
Yep, thanks to Andrej's review, I saw that I was using wrong resource reference.